book/content/boundary_conditions/concentration.md

```diff
@@ -0,0 +1,169 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.18.1
+kernelspec:
+  display_name: festim-workshop
+  language: python
+  name: python3
+---
+
+# Concentration #
+
+Boundary conditions (BCs) are essential to FESTIM simulations, as they describe the mathematical problem at the boundaries of the simulated domain. Read more about BCs _[here](https://festim.readthedocs.io/en/fenicsx/userguide/boundary_conditions.html)_.
+
+This tutorial goes over how to define concentration boundary conditions for hydrogen transport simulations.
+
+Objectives:
+* Define a fixed concentration BC
+* Choose which solubility law (Sieverts' or Henry's)
+* Solve a hydrogen transport problem with plasma implantation
+
++++
+
+## Defining fixed concentration ##
+
+BCs need to be assigned to surfaces using FESTIM's `SurfaceSubdomain` class. To define the concentration of a specific species, we can use `FixedConcentrationBC`:
+
+```{code-cell} ipython3
+from festim import FixedConcentrationBC, Species, SurfaceSubdomain
+
+boundary = SurfaceSubdomain(id=1)
+H = Species(name="Hydrogen")
+
+my_bc = FixedConcentrationBC(subdomain=boundary, value=10, species=H)
+```
+
+The imposed concentration can be dependent on space, time and temperature, such as:
+
+$$ 
+
+BC = 10 + x^2 + T(t+2)
+
+$$
+
+```{code-cell} ipython3
+my_custom_value = lambda x, t, T: 10 + x[0]**2 + T *(t + 2)
+
+my_bc = FixedConcentrationBC(subdomain=boundary, value=my_custom_value, species=H)
+```
+
+## Choosing a solubility law ##
+
+Users can define the surface concentration using either Sieverts’ law, $c = S(T)\sqrt P$, or Henry's law, $c=K_H P$, where $S(T)$ and $K_H$ denote the temperature-dependent Sieverts’ and Henry’s solubility coefficients, respectively, and $P$ is the partial pressure of the species on the surface. 
+
+For Sieverts' law of solubility, we can use `festim.SievertsBC`:
+
+```{code-cell} ipython3
+from festim import SievertsBC, SurfaceSubdomain, Species
+
+boundary = SurfaceSubdomain(id=1)
+H = Species(name="Hydrogen")
+
+custom_pressure_value = lambda t: 2 + t
+my_bc = SievertsBC(subdomain=3, S_0=2, E_S=0.1, species=H, pressure=custom_pressure_value)
+```
+
+Similarly, for Henry's law of solubility, we can use `festim.HenrysBC`:
+
+```{code-cell} ipython3
+from festim import HenrysBC
+
+pressure_value = lambda t: 5 * t
+my_bc = HenrysBC(subdomain=3, H_0=1.5, E_H=0.2, species=H, pressure=pressure_value)
+```
+
+## Plasma implantation approximation ##
```

Comment:
There's actually an issue here: this is not a boundary condition approximation, you set an actual volumetric source.